File: solutions/0238-product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        prefix = [0 for i in range (len(nums))]
        postfix = [0 for i in range (len(nums))]
        result = [0 for i in range(len(nums))]

        temp = 1
        for i in range (len(nums)):
            temp *= nums[i]  
            prefix[i] = temp

        temp = 1
        for i in range (len(nums) - 1, 0, -1):
            temp *= nums[i]
            postfix[i] = temp

        for i in range(len(nums)):
            if (i == 0):
                result[i] = postfix [i + 1] 
            elif (i == len(nums) - 1):
                result[i] = prefix [i -1]
            else:        
                result[i] = prefix[i - 1] * postfix [i + 1] 

        return result        

File: solutions/0238-product-of-array-except-self/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
    let fArr = []
    let start = 1

    for (n of nums) {
        fArr.push(start)
        start = start * n
    }

    let bArr = []
    start = 1

    for (let i = nums.length - 1; i >= 0; i--) {
        bArr[i] = start
        start = start * nums[i]
    }

    let result = []

    for (let i = 0; i < nums.length; i++) {
        result[i] = fArr[i] * bArr[i]
    }

    return result
};
